src/read_data.c

```diff
@@ -34,7 +34,9 @@
 #if HAVE_ZLIB
 #include <zlib.h>
 #endif
+#include <float.h>
 #include <stddef.h>
+#include <stdint.h>
```

Comment:
```suggestion
#include <limits.h>
```

This fails if `HAVE_STDINT_H` is not defined. Did you mean to include limits.h instead?

Reply:
Oh. Doesn't matio require C99? stdint.h I believe is the correct header for `UINT32_MAX` and friends: https://en.cppreference.com/w/c/types/integer

Reply:
No, it's not C99. Go for https://en.cppreference.com/w/c/types/limits instead.

Reply:
>No, it's not C99.

Oh. So I guess that's why you have `mat_int32_t` instead of just using `int32_t`?

>Go for https://en.cppreference.com/w/c/types/limits instead.

`UINT32_MAX` is not there.

So I guess we need a custom `MAT_UINT32_MAX` somewhere...

Reply:
Or I just use literals like `4294967295U`?

Reply:
You can see the safe-math.h where literals (in hex) are used.